File: bench/micro/form_hashmap_set.py
"""Python equivalent of form_hashmap_set.ap."""

import time


class Entry:
    __slots__ = ("id", "v")
    def __init__(self, id, v):
        self.id = id
        self.v = v


n = 1_000_000
m = {}
t0 = time.monotonic_ns()
for i in range(n):
    m[i] = Entry(i, i + 1)
elapsed = time.monotonic_ns() - t0
print(f"n={n}")
print(f"len={len(m)}")
print(f"elapsed_ns={elapsed}")

File: bench/micro/form_hashmap_set.go
// Go equivalent of form_hashmap_set.ap.
// N inserts into a map[int]Entry. Go's map is hash-based with
// chained-bucket collision resolution — same big-O as Aperio's
// open-addressing implementation, different layout.
package main

import (
	"fmt"
	"time"
)

type Entry struct {
	id int
	v  int
}

func main() {
	n := 1000000
	m := make(map[int]Entry)
	t0 := time.Now()
	for i := 0; i < n; i++ {
		m[i] = Entry{id: i, v: i + 1}
	}
	elapsed := time.Since(t0).Nanoseconds()
	fmt.Printf("n=%d\n", n)
	fmt.Printf("len=%d\n", len(m))
	fmt.Printf("elapsed_ns=%d\n", elapsed)
}
